code/clue.js

```diff
@@ -1,59 +1,361 @@
+      document.getElementById(loadID.id).style.opacity = 0;
+      mystery.weapon = randomSelector(weaponsIncreased);
+      document.getElementById('weaponName').style.background = mystery.weapon.color;
+      document.getElementById('weaponName').innerHTML = `Type: ${mystery.weapon.name}`;
+      document.getElementById('weaponWeight').innerHTML = `Weight: ${mystery.weapon.weight}`;
+      weaponPicked = true;
+      checkScroll();
+    }, 2000);
+  } else if (showInfoMessage("pickWeapon"));
+}
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+const pickRoom = (loadID) => {
+  if (killerPicked && weaponPicked) {
+    document.getElementById(loadID.id).style.opacity = 1;
+    setTimeout(function () {
+      document.getElementById(loadID.id).style.opacity = 0;
+      const increasedRoom = increaseChanceForRoom(mystery.weapon);
+      mystery.room = randomSelector(increasedRoom);
+      document.getElementById('roomName').innerHTML = mystery.room;
+      roomPicked = true;
+      checkScroll();
+    }, 2000);
+  } else(showInfoMessage("pickRoom"));
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+const showInfoMessage = (callerFunctionName) => {
+  if (callerFunctionName === "pickWeapon" || (callerFunctionName === "pickRoom" && !killerPicked)) {
+    document.getElementById("killerInfoMessage").style.opacity = 1;
+    setTimeout(function () {
+      document.getElementById("killerInfoMessage").style.opacity = 0;
+    }, 2000);
+  } else if (callerFunctionName === "pickRoom" && killerPicked) {
+    document.getElementById("weaponInfoMessage").style.opacity = 1;
+    setTimeout(function () {
+      document.getElementById("weaponInfoMessage").style.opacity = 0;
+    }, 2000);
+  } else if (callerFunctionName === "revealMystery") {
+    document.getElementById("mysteryInfoMessage").style.opacity = 1;
+    setTimeout(function () {
+      document.getElementById("mysteryInfoMessage").style.opacity = 0;
+    }, 2000);
+  }
+}
+
+const shuffleFavoriteWeapon = () => {
+  suspects.forEach(suspect => {
+    const selectedRandomWeapon = randomSelector(weapons).id;
+    suspect.favoriteWeapon = selectedRandomWeapon;
+  });
+}
+
+const increaseChanceForFavWeapon = (killerObject) => {
+  let weaponsIncreased = new Array();
+  const preferredWeapon = killerObject.favoriteWeapon;
+  const weaponObject = weapons.find(weapon => weapon.id === preferredWeapon);
+  const increaseChance = weaponObject.increasedChanceOfDraw;
+  for (i = 0; i < increaseChance; i++) {
+    weaponsIncreased.push(weaponObject);
+  }
+  weaponsIncreased = weaponsIncreased.concat(weapons);
+  return weaponsIncreased;
+}
```

Comment:
Really beautiful.  Only one small suggestion `const increasedChance` instead of `const increaseChance`.  Most likely you intended that already.